src/core/core.scale.js

```diff
@@ -553,25 +528,22 @@ var Scale = Element.extend({
 		helpers.callback(this.options.beforeBuildTicks, [this]);
 	},
 	buildTicks: helpers.noop,
-	afterBuildTicks: function(ticks) {
-		var me = this;
-		// ticks is empty for old axis implementations here
-		if (isArray(ticks) && ticks.length) {
-			return helpers.callback(me.options.afterBuildTicks, [me, ticks]);
-		}
-		// Support old implementations (that modified `this.ticks` directly in buildTicks)
-		me.ticks = helpers.callback(me.options.afterBuildTicks, [me, me.ticks]) || me.ticks;
-		return ticks;
+	afterBuildTicks: function() {
+		helpers.callback(this.options.afterBuildTicks, [this]);
 	},
 
 	beforeTickToLabelConversion: function() {
 		helpers.callback(this.options.beforeTickToLabelConversion, [this]);
 	},
-	convertTicksToLabels: function() {
+	/**
+	 * Convert ticks to label strings
+	 */
+	convertTicksToLabels: function(ticks) {
```

Comment:
I'd rename this to `generateTickLabels` or equivalent, and directly set the label property of tick.